services/shared/src/main/java/com/wedriveu/services/shared/rabbitmq/VerticlePublisher.java

```diff
@@ -17,42 +14,20 @@
  */
 public class VerticlePublisher extends AbstractVerticle {
 
-    private static RabbitMQClient client;
-    private static String TAG = VerticlePublisher.class.getSimpleName();
+    private RabbitMQClient client;
 
     @Override
     public void start(Future<Void> startFuture) throws Exception {
-        super.start(startFuture);
+        client = RabbitMQFactory.createClient(vertx);
+        client.start(startFuture.completer());
     }
 
     protected void publish(String exchangeName, String routingKey, JsonObject data) {
-        client = RabbitMQConfig.getInstance(vertx).getRabbitMQClient();
-        client.start(onStartCompleted -> {
-                    if (onStartCompleted.succeeded()) {
-                        declareExchanges(onDeclareCompleted -> {
-                            if (onDeclareCompleted.succeeded()) {
-                                publishToConsumer(exchangeName, routingKey, data);
-                            } else {
-                                Log.error(TAG, onDeclareCompleted.cause().getMessage(), onDeclareCompleted.cause());
-                            }
-                        });
-                    }
-                }
-        );
-    }
-
-    protected void declareExchanges(Handler<AsyncResult<Void>> handler) {
-        client.exchangeDeclare(VEHICLE_SERVICE_EXCHANGE,
-                EXCHANGE_TYPE,
-                false,
-                false,
-                handler);
-    }
-
-    protected void publishToConsumer(String exchangeName,
-                                     String routingKey,
-                                     JsonObject dataToUser) {
-        client.basicPublish(exchangeName, routingKey, dataToUser, null);
+        client.basicPublish(exchangeName, routingKey, data, onPublish -> {
```

Comment:
Can't we pass an handler or a future to this `publish` method to let also the subclass do something after publishing if necessary?